cmd/hooks/wait-paths/main.go

```diff
@@ -0,0 +1,72 @@
+// +build linux
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	pathsFlag   = "paths"
+	timeoutFlag = "timeout"
+)
+
+// This is a hook that waits for a specific path to appear.
+// The hook has required list of comma-separated paths and a default timeout in seconds.
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "wait-paths"
+	app.Usage = "Provide a list paths and an optional timeout"
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:     pathsFlag + ",p",
+			Usage:    "Comma-separated list of paths that should become available",
+			Required: true,
+		},
+		cli.IntFlag{
+			Name:  timeoutFlag + ",t",
+			Usage: "Timeout in seconds",
+			Value: 30,
+		},
+	}
+	app.Action = run
+	if err := app.Run(os.Args); err != nil {
+		logrus.Errorf("%s", err)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func run(cCtx *cli.Context) error {
+	timeout := cCtx.GlobalInt(timeoutFlag)
+	paths := strings.Split(cCtx.GlobalString(pathsFlag), ",")
+
+	waitCtx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
+	defer cancel()
+
+	for _, path := range paths {
```

Comment:
do we want to wait for each path in parallel?

Reply:
I don't think it will matter, since `Stat` should be fast and the bottleneck would really be the path/mount that takes longest.

Reply:
Yeah that's probably okay especially since this isn't a critical path area. Resolving.